pkg/common/utils.go

```diff
@@ -479,12 +474,49 @@ func existingErrorCode(err error) (codes.Code, error) {
 	if err == nil {
 		return codes.Unknown, fmt.Errorf("null error")
 	}
-	if status, ok := status.FromError(err); ok {
-		return status.Code(), nil
+	var tmpError *TemporaryError
+	// This explicitly checks our error is a temporary error before extracting its
+	// status, as there can be other errors that can qualify as statusable
+	// while not necessarily being temporary.
+	if errors.As(err, &tmpError) {
+		if status, ok := status.FromError(err); ok {
+			return status.Code(), nil
+		}
+	}
+	// We want to make sure we catch other temporary error types that are statusable.
```

Comment:
Nit: Remove the word temporary. I think this is easy to confuse with the temporary error above, and we don't mean temporary like TemporaryError here.

 We want to make sure we catch other temporary error types that are statusable. -> We want to make sure we catch other error types that are statusable.